catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchParserImpl.java

```diff
@@ -132,31 +123,35 @@ public static void populateSearchOptions(WebClient client, Query query, Subject
         String filterStr = "";
         String sortStr = null;
 
-        if (query != null) {
 
-            maxPerPage = String.valueOf(query.getPageSize());
-            if (query.getPageSize() > DEFAULT_TOTAL_MAX) {
-                maxTotalSize = maxPerPage;
-            } else if (query.getPageSize() <= 0) {
-                maxTotalSize = String.valueOf(DEFAULT_TOTAL_MAX);
-            }
+        if (queryRequest != null) {
+            Query query = queryRequest.getQuery();
+
+            if (query != null) {
+                maxPerPage = String.valueOf(query.getPageSize());
+                if (query.getPageSize() > DEFAULT_TOTAL_MAX) {
+                    maxTotalSize = maxPerPage;
+                } else if (query.getPageSize() <= 0) {
+                    maxTotalSize = String.valueOf(DEFAULT_TOTAL_MAX);
+                }
 
-            start = Integer.toString(query.getStartIndex());
+                start = Integer.toString(query.getStartIndex());
 
-            timeout = Long.toString(query.getTimeoutMillis());
+                timeout = Long.toString(query.getTimeoutMillis());
 
-            sortStr = translateToOpenSearchSort(query.getSortBy());
+                sortStr = translateToOpenSearchSort(query.getSortBy());
 
-            if (subject != null && subject.getPrincipals() != null && !subject.getPrincipals()
-                    .isEmpty()) {
-                List principals = subject.getPrincipals()
-                        .asList();
-                for (Object principal : principals) {
-                    if (principal instanceof SecurityAssertion) {
-                        SecurityAssertion assertion = (SecurityAssertion) principal;
-                        Principal assertionPrincipal = assertion.getPrincipal();
-                        if (assertionPrincipal != null) {
-                            dn = assertionPrincipal.getName();
+                if (subject != null && subject.getPrincipals() != null && !subject.getPrincipals()
+                        .isEmpty()) {
+                    List principals = subject.getPrincipals()
+                            .asList();
+                    for (Object principal : principals) {
+                        if (principal instanceof SecurityAssertion) {
+                            SecurityAssertion assertion = (SecurityAssertion) principal;
+                            Principal assertionPrincipal = assertion.getPrincipal();
+                            if (assertionPrincipal != null) {
+                                dn = assertionPrincipal.getName();
```

Comment:
Should dn be assigned multiple times as that seems possible here.
The `findFirst` stream option above would resolve this issue too, if it's a problem.